samples/unit-tests/series/datalabels/demo.js

```diff
@@ -381,6 +386,19 @@ QUnit.test('Datalabels overlap in hidden series (#3866)', function (assert) {
         ),
         'All six labels of the second series should be visible.'
     );
+
+    assert.ok(
+        true,
+        'No console error should be displayed (#19457).'
+    );
+
+    assert.strictEqual(
+        chart.options.plotOptions.series.dataLabels.filter(({ enabled }) =>
+            enabled
+        ).length,
+        chart.series[0].points[0].dataLabels.length,
+        'Only the second point\'s data label should be rendered (#19457).'
+    );
```

Comment:
It would be better to move these tests to test, which is responsible for multiple data labels: `unit-tests/datalabels/multiple-labels`. There are already declared data labels as an array, so one update there should be fine.

Reply:
Agree, it's cleaner now as there's no need to add any extra chart config, thanks!